### PR TITLE
allow plugins to register helpSteps #442

### PR DESCRIPTION
**Pull request by patrick-austin**
## Description
Add optional `helpSteps` to the plugin payload and register the contents for the tour. Retain the previous `helpText` (which only applies to the plugin in the navbar itself) to maintain compatibility if the plugins haven't been updated.

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Travis build
- [ ] Review changes to test coverage
- [ ] Check that the `helpText` still displays for plugins using that method of registering tour steps

Testing the new `helpSteps` will require code from ral-facilities/datagateway#430 (PR ral-facilities/datagateway#442)

## Agile board tracking
connect to #442 ral-facilities/datagateway#430

### CODE REVIEW

File: src/state/actions/scigateway.actions.test.tsx [hunk 1]
```diff
@@ -422,6 +422,7 @@ describe('scigateway actions', () => {
     );
 
     localStorage.setItem('darkMode', 'true');
+    console.log(localStorage.getItem('darkMode'));
```
Comment (louise-davies) [inline src/state/actions/scigateway.actions.test.tsx:425]: remove logging statements

File: src/state/actions/scigateway.actions.test.tsx [hunk 2]
```diff
@@ -430,6 +431,7 @@ describe('scigateway actions', () => {
 
     await asyncAction(dispatch, getState);
 
+    console.log(actions);
```
Comment (louise-davies) [inline src/state/actions/scigateway.actions.test.tsx:434]: same here

File: src/state/actions/scigateway.actions.tsx [hunk 3]
```diff
@@ -149,8 +149,10 @@ export const configureSite = (): ThunkResult<Promise<void>> => {
     // load dark mode preference from local storage into store
     // otherwise, fetch system preference
     const darkModeLocalStorage = localStorage.getItem('darkMode');
+    console.log(darkModeLocalStorage);
     if (darkModeLocalStorage) {
       const preference = darkModeLocalStorage === 'true' ? true : false;
+      console.log(preference);
```
Comment (louise-davies) [inline src/state/actions/scigateway.actions.tsx:155]: remove

**Review by louise-davies — CHANGES_REQUESTED** [review]

**Comment by patrick-austin** [comment]
@louise-davies I'll get rid of the logging, with the failing darkmode test should I skip it or leave it in? Assuming it's still failing in the Travis runs.

**Comment by louise-davies** [comment]
@patrick-austin I'd say just skip it, add in a TODO to investigate and perhaps we should create an issue on the repo? Just to track it.

**Review by louise-davies — APPROVED** [review]

**Pull request merged**